### DISCUSSION TITLE
Should `create-next-app` default to using ESLint?

### DISCUSSION THREAD

**Discussion by leerob** [vercel/next.js#73905] [Polls]
Hey y'all. We're considering switching the default of `create-next-app` to [not use ESLint](https://github.com/vercel/next.js/pull/73864).

One reason to remove would be: fewer configuration files, and less things to remove. Notably, you can still pick ESLint during the `create-next-app` setup – it just wouldn't be the default. Another related bit is that we'd like to [also support Biome](https://github.com/vercel/next.js/discussions/59347) for `next lint` in the future.

One reason to keep ESLint as the default is that it gives you default checks around React development (e.g. hook usage), which could help beginners who are learning to build with React and Next.js.

**Comment by JClackett** [comment]
Biome has rules of hooks I believe? Or maybe just the dependency array checks

**Comment by MartinCura** [comment]
I think having a linter is a more sane default than not, especially in a React project. If Biome can do similar checks with hooks and such, then Eslint shouldn't necessarily be the first choice when choosing in the interactive CLI.

But if someone's running `create-next-app` non-interactively, including _some_ linter is the sane default for me. => Voted "Yes".

**Reply by clicktodev** [reply]
I've worked with enough bad code and broken useEffects to know that i never want to touch a react codebase without eslint again.
Voted YES

**Comment by philwolstenholme** [comment]
For someone without a strong opinion on linters then they're better off with ESLint than with no linter at all. The ESLint ecosystem is also still larger so it seems a more sensible default. 

Anyone who is opinionated enough to prefer Biome or prefer no linter at all is probably also experienced enough to remove ESLint and set up things the way they want them without this being too much of a big deal. 

What's wrong with config files other than people ([and CEOs](https://x.com/rauchg/status/1867289170731045244)…) moaning about them on Twitter? I would prefer them all to be hidden away in a [`.config` directory](https://github.com/pi0/config-dir) but I think that'll come eventually.

**Reply by leerob** [reply]
It's not just that config files bloat your directory, but also that they add additional things that can break between version upgrades (versus zero-config where that is placed on the library or tool author). Zero-config is the dream.

I don't know if `.config` directory is the right solution, but rather than tools try to be zero config (thank you Tailwind!)

**Reply by philwolstenholme** [reply]
That's a really good point actually, I hadn't thought about how it must be a pain to redo the config files when dependencies make breaking changes. A config directory won't help with that at all, like you say.

**Reply by bitttttten** [reply]
I have definitely learned techniques, ways of thinking, and improved my JavaScript-grammar with linters. It also can just catch bugs for me, before even leaving my browser. I think for that reason it will help beginners more.

Once con is that more files in a repo is more overwhelming for beginners, although this is very subjective as some beginners may just ignore all the "extra files" and don't mind.

It's a tough call.

**Comment by leerob** [comment]
Classic 50/50 poll response so far 😂

**Comment by Arctomachine** [comment]
I vote to support biome as default linter. Have been using it in prod for year or two already, but mostly as formatter because next uses eslint internally.
Also will there be any migration guide for old projects to switch eslint to biome in build script? Or is it possible already manually?

**Comment by clicktodev** [comment]
Consider reverting this PR please https://github.com/vercel/next.js/pull/73864
I think this was a hasty decision and i suggest a new poll be created with a big enough sample size to warrant such a non beginner friendly change.

**Comment by ValentinGurkov** [comment]
Another vote for biome!

**Comment by tausifcreates** [comment]
Bring back eslint!
